File: problem7.py
import math
def is_prime(num):
  divisor = math.floor(math.sqrt(num))
  while (divisor >= 2):
    if (num % divisor == 0):
      return False
    divisor -= 1
  return True

def nth_prime(count, num = 2):
  while (count > 0):
    if (is_prime(num)):
      count -= 1
    num += 1
  return num - 1

print nth_prime(10001)

File: ruby/problem7.rb
def is_prime?(num)
  i = 2
  while i <= Math.sqrt(num)
  return false if num % i == 0
    i += 1
  end
  return true
end

def ten_thousand_first_prime
  i = 3
  count = 0
  while true
    if is_prime?(i)
      count += 1
    end
    break if count == 10_000
  	i += 2
  end
  i

end

p ten_thousand_first_prime
